# Milestone 1 — Inspeção Inicial dos Dados

Nesta fase é realizada a primeira exploração do conjunto de dados Telco Customer Churn, com o objetivo de compreender a sua estrutura, dimensão, tipos de variáveis, valores em falta, duplicados e distribuição da variável Churn.

In [12]:
import pandas as pd
import numpy as np
from pathlib import Path

csv_files = list(Path('/kaggle/input').rglob('*.csv'))

print("Ficheiros CSV encontrados:")
for file in csv_files:
    print(file)

df = pd.read_csv(csv_files[0])

Ficheiros CSV encontrados:
/kaggle/input/datasets/blastchar/telco-customer-churn/WA_Fn-UseC_-Telco-Customer-Churn.csv


## Visualização inicial dos dados

Nesta secção são apresentadas as primeiras linhas do dataset e a sua dimensão, para perceber a estrutura geral da tabela.

In [13]:
display(df.head())

print(f"Número de linhas: {df.shape[0]}")
print(f"Número de colunas: {df.shape[1]}")

,customerID,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,MultipleLines,InternetService,OnlineSecurity,...,DeviceProtection,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,MonthlyCharges,TotalCharges,Churn
0,7590-VHVEG,Female,0,Yes,No,1,No,No phone service,DSL,No,...,No,No,No,No,Month-to-month,Yes,Electronic check,29.85,29.85,No
1,5575-GNVDE,Male,0,No,No,34,Yes,No,DSL,Yes,...,Yes,No,No,No,One year,No,Mailed check,56.95,1889.5,No
2,3668-QPYBK,Male,0,No,No,2,Yes,No,DSL,Yes,...,No,No,No,No,Month-to-month,Yes,Mailed check,53.85,108.15,Yes
3,7795-CFOCW,Male,0,No,No,45,No,No phone service,DSL,Yes,...,Yes,Yes,No,No,One year,No,Bank transfer (automatic),42.30,1840.75,No
4,9237-HQITU,Female,0,No,No,2,Yes,No,Fiber optic,No,...,No,No,No,No,Month-to-month,Yes,Electronic check,70.70,151.65,Yes


Número de linhas: 7043
Número de colunas: 21


## Inspeção geral e verificação inicial da qualidade dos dados

Nesta célula são analisados os tipos de dados, estatísticas descritivas, valores nulos, duplicados, distribuição da variável Churn e possíveis problemas na coluna TotalCharges.

In [16]:
print("\nInformação geral do dataset:")
df.info()

print("\nEstatística descritiva das variáveis numéricas:")
display(df.describe())

print("\nEstatística descritiva das variáveis categóricas:")
display(df.describe(include='object'))

print("\nValores nulos por coluna:")
display(df.isnull().sum().to_frame("Valores nulos"))

print("\nNúmero de linhas duplicadas:")
print(df.duplicated().sum())

print("\nDistribuição da variável Churn:")
display(df['Churn'].value_counts().to_frame("Contagem"))

print("\nDistribuição percentual da variável Churn:")
display((df['Churn'].value_counts(normalize=True) * 100).round(2).to_frame("Percentagem (%)"))

print("\nTipo de dados da coluna TotalCharges:")
print(df['TotalCharges'].dtype)

# Conversão temporária da coluna TotalCharges para verificar problemas
df['TotalCharges_num'] = pd.to_numeric(df['TotalCharges'], errors='coerce')

print("\nNúmero de valores problemáticos/em falta após conversão de TotalCharges:")
print(df['TotalCharges_num'].isnull().sum())

print("\nLinhas com TotalCharges problemático:")
display(df[df['TotalCharges_num'].isnull()])


Informação geral do dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 22 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        7043 non-null   object 
 1   gender            7043 non-null   object 
 2   SeniorCitizen     7043 non-null   int64  
 3   Partner           7043 non-null   object 
 4   Dependents        7043 non-null   object 
 5   tenure            7043 non-null   int64  
 6   PhoneService      7043 non-null   object 
 7   MultipleLines     7043 non-null   object 
 8   InternetService   7043 non-null   object 
 9   OnlineSecurity    7043 non-null   object 
 10  OnlineBackup      7043 non-null   object 
 11  DeviceProtection  7043 non-null   object 
 12  TechSupport       7043 non-null   object 
 13  StreamingTV       7043 non-null   object 
 14  StreamingMovies   7043 non-null   object 
 15  Contract          7043 non-null   object 
 16  PaperlessBil

,SeniorCitizen,tenure,MonthlyCharges,TotalCharges_num
count,7043.000000,7043.000000,7043.000000,7032.000000
mean,0.162147,32.371149,64.761692,2283.300441
std,0.368612,24.559481,30.090047,2266.771362
min,0.000000,0.000000,18.250000,18.800000
25%,0.000000,9.000000,35.500000,401.450000
50%,0.000000,29.000000,70.350000,1397.475000
75%,0.000000,55.000000,89.850000,3794.737500
max,1.000000,72.000000,118.750000,8684.800000



Estatística descritiva das variáveis categóricas:


,customerID,gender,Partner,Dependents,PhoneService,MultipleLines,InternetService,OnlineSecurity,OnlineBackup,DeviceProtection,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,TotalCharges,Churn
count,7043,7043,7043,7043,7043,7043,7043,7043,7043,7043,7043,7043,7043,7043,7043,7043,7043,7043
unique,7043,2,2,2,2,3,3,3,3,3,3,3,3,3,2,4,6531,2
top,7590-VHVEG,Male,No,No,Yes,No,Fiber optic,No,No,No,No,No,No,Month-to-month,Yes,Electronic check,,No
freq,1,3555,3641,4933,6361,3390,3096,3498,3088,3095,3473,2810,2785,3875,4171,2365,11,5174



Valores nulos por coluna:


,Valores nulos
customerID,0
gender,0
SeniorCitizen,0
Partner,0
Dependents,0
tenure,0
PhoneService,0
MultipleLines,0
InternetService,0
OnlineSecurity,0



Número de linhas duplicadas:
0

Distribuição da variável Churn:


,Contagem
Churn,
No,5174
Yes,1869



Distribuição percentual da variável Churn:


,Percentagem (%)
Churn,
No,73.46
Yes,26.54



Tipo de dados da coluna TotalCharges:
object

Número de valores problemáticos/em falta após conversão de TotalCharges:
11

Linhas com TotalCharges problemático:


,customerID,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,MultipleLines,InternetService,OnlineSecurity,...,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,MonthlyCharges,TotalCharges,Churn,TotalCharges_num
488,4472-LVYGI,Female,0,Yes,Yes,0,No,No phone service,DSL,Yes,...,Yes,Yes,No,Two year,Yes,Bank transfer (automatic),52.55,,No,NaN
753,3115-CZMZD,Male,0,No,Yes,0,Yes,No,No,No internet service,...,No internet service,No internet service,No internet service,Two year,No,Mailed check,20.25,,No,NaN
936,5709-LVOEQ,Female,0,Yes,Yes,0,Yes,No,DSL,Yes,...,No,Yes,Yes,Two year,No,Mailed check,80.85,,No,NaN
1082,4367-NUYAO,Male,0,Yes,Yes,0,Yes,Yes,No,No internet service,...,No internet service,No internet service,No internet service,Two year,No,Mailed check,25.75,,No,NaN
1340,1371-DWPAZ,Female,0,Yes,Yes,0,No,No phone service,DSL,Yes,...,Yes,Yes,No,Two year,No,Credit card (automatic),56.05,,No,NaN
3331,7644-OMVMY,Male,0,Yes,Yes,0,Yes,No,No,No internet service,...,No internet service,No internet service,No internet service,Two year,No,Mailed check,19.85,,No,NaN
3826,3213-VVOLG,Male,0,Yes,Yes,0,Yes,Yes,No,No internet service,...,No internet service,No internet service,No internet service,Two year,No,Mailed check,25.35,,No,NaN
4380,2520-SGTTA,Female,0,Yes,Yes,0,Yes,No,No,No internet service,...,No internet service,No internet service,No internet service,Two year,No,Mailed check,20.00,,No,NaN
5218,2923-ARZLG,Male,0,Yes,Yes,0,Yes,No,No,No internet service,...,No internet service,No internet service,No internet service,One year,Yes,Mailed check,19.70,,No,NaN
6670,4075-WKNIU,Female,0,Yes,Yes,0,Yes,Yes,DSL,No,...,Yes,Yes,No,Two year,No,Mailed check,73.35,,No,NaN


## Síntese da inspeção inicial

A inspeção inicial permitiu confirmar que o dataset tem 7043 linhas e 21 colunas. Também foram analisados os tipos de dados, valores nulos, duplicados e a distribuição da variável Churn. Verificou-se ainda que a coluna TotalCharges se encontra como variável categórica, apesar de representar valores monetários, pelo que deverá ser tratada numa fase posterior.